platform-common/src/main/java/com/softicar/platform/common/io/resource/supplier/IResourceSupplier.java

```diff
@@ -28,6 +32,22 @@
 	 */
 	IResourceKey getResourceKey();
 
+	/**
+	 * Returns the content of the {@link IResource} as UTF-8 encoded text.
+	 * <p>
+	 * May only be called on textual resources.
+	 *
+	 * @return the content as UTF-8 encoded text (never <i>null</i>)
+	 */
+	default String getTextUtf8() {
```

Comment:
uh, I think this should rather be implemented on IResource, not the supplier

Reply:
indeed